docs/internals.rst

```diff
@@ -0,0 +1,332 @@
+
+That’s the reason why even if you see ``node-gyp`` errors and failures during the installation process, afterwards Dredd seems to normally work and correctly parses API Blueprint documents.
+
+
+Avoiding C++ compilation
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``--no-optional`` option forces the JavaScript version of Drafter and avoids any compilation attempts when installing Dredd:
+
+.. code-block:: shell
+
+   $ npm install dredd --global --no-optional
+
+
+Troubleshooting C++ compilation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you need the performance of the C++11 parser, but you are struggling to get it installed, it’s usually because of the following problems:
+
+-  **Your machine is missing a C++11 compiler.** See how to fix this on `Windows <https://github.com/apiaryio/drafter/wiki/Building-on-Windows>`__ or `Travis CI <https://github.com/apiaryio/protagonist/blob/master/.travis.yml>`__.
+-  **npm was used with Python 3.** ``node-gyp``, which performs the compilation, doesn’t support Python 3 yet. If your default Python is 3 (see ``python --version``), `tell npm to use an older version <https://stackoverflow.com/a/22433804/325365>`__.
+
+
+Supported Node.js versions
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Given the `table with LTS schedule <https://github.com/nodejs/Release>`__, only versions marked as **Current**, **Maintenance**, or **Active** are supported, until their **Maintenance End**. The testing matrix of Dredd’s CI builds must contain all currently supported versions and must not contain any unsupported versions. The same applies for the underlying libraries, such as `Dredd Transactions <https://github.com/apiaryio/dredd-transactions>`__ or `Gavel.js <https://github.com/apiaryio/gavel.js/>`__. In ``appveyor.yml`` the latest supported Node.js version should be used.
+
+When dropping support for a certain Node.js version, it should be removed from the testing matrix, and it **must** be delivered as a breaking change, which increments Dredd's major version number.
+
+
+Dependencies
+~~~~~~~~~~~~
+
+New versions of dependencies are monitored by `David <https://david-dm.org/apiaryio/dredd>`__ and `Greenkeeper <https://greenkeeper.io/>`__. Vulnerabilities are monitored by `Snyk <https://snyk.io/test/npm/dredd>`__.
+
+Dependencies should not be specified in a loose way - only exact versions are allowed. This is ensured by ``.npmrc`` and the lock file. Any changes to dependencies (version upgrades included) are a subject to internal policies and must be first checked and approved by the maintainers before merged to ``master``. This is because we are trying to be good Open Source citizens and to do our best to comply with licenses of all our dependencies.
```

Comment:
I wonder, does it make sense to enable FOSSA? FWIU it is free for OSS to some extent. At least then there is a status badge on PRs indicating if the license of a dependency is problematic so people are aware upfront.

Reply:
I'd like to have a single source of truth. FOSSA is nice, but could be misleading as it could have different results or opinions than Oracle's internal processes. Since @w-vi's squad is internally working on a tool to provide some information under PRs, I'd rather wait for that one and to lobby for the tool to be Open-Source-repos-compatible.

Reply:
OSS repos compatibility is in our mind so this should be covered.